scripts/utils/celltype-plot-helper-functions.R

```diff
@@ -0,0 +1,140 @@
+# Helper functions for creating cell type plots 
+# These functions are taken directly from `scpca-nf/templates/celltypes_qc.rmd`,
+# `scpca-nf/templates/celltypes_supplemental_report.rmd`, and `scpca-nf/templates/utils/celltype_functions.Rmd`
+
+library(SingleCellExperiment)
+
+# Set up cell type dataframe ---------------------------------------------------
+
+#' Create `celltype_df` data frame for use in cell type QC reports
```

Comment:
I'd get rid of this function (unless we need it for other plots in the paper?) and just include this in the Fig4B script directly - 

Actually, I'd just get rid of this function - unless we'll need it for another plot in the paper? - and just put this in the Fig4B script directly (with proper spacing)

```
celltype_df <- processed_sce |>
  scuttle::makePerCellDF(use.dimred = "UMAP") |>
  dplyr::select(
     UMAP1 = UMAP.1, 
     UMAP2 = UMAP.2
     singler_celltype_annotation, 
     cellassign_celltype_annotation
    ) |>
     dplyr::filter(
       singler_celltype_annotation != "Unclassified cell",
        cellassign_celltype_annotation != "Unclassified cell"
      ) |> 
      prepare_automated_annotation_values(singler_celltype_annotation)
      prepare_automated_annotation_values(cellassign_celltype_annotation)
```